### PR TITLE
fix(gatsby): Improve error message when calling useStaticQuery without graphql

### PR DESCRIPTION
**Pull request by blainekasten**
## Description

I ran into a little hiccup trying to use `useStaticQuery`, I forgot that I had to wrap my string in the `graphql` call. That's fine, but the error message was problematic. This fix makes the following change:

**Before**
image

**After**
image

### CODE REVIEW

File: packages/gatsby/cache-dir/gatsby-browser-entry.js [hunk 1]
```diff
@@ -57,6 +57,19 @@ const useStaticQuery = query => {
     )
   }
   const context = React.useContext(StaticQueryContext)
+
+  // query is a stringified number like `3303882` when wrapped with graphql, If a user forgets
+  // to wrap the query in a grqphql, then casting it to a Number results in `NaN` allowing us to
+  // catch the misuse of the API and give proper direction
+  if (isNaN(Number(query))) {
```
Comment (wardpeet) [inline packages/gatsby/cache-dir/gatsby-browser-entry.js:64]: Would it be ok to only show this error in development? Error messages are "expensive" in production code. @blainekasten

Comment (blainekasten) [reply packages/gatsby/cache-dir/gatsby-browser-entry.js:64]: Hmm good question. Would this ever really leak to production?

Comment (blainekasten) [reply packages/gatsby/cache-dir/gatsby-browser-entry.js:64]: Is it just a request to wrap it in a `process.env.NODE_ENV !== 'production'` ?

**Comment by vladar** [comment]
Published in `gatsby@2.20.29`

**Pull request merged**